code/hud/hudconfig.cpp

```diff
@@ -1323,10 +1323,8 @@ void hud_config_button_do(int n)
 			break;
 		}
 
-		if (HC_current_file >= (int)HC_preset_filenames.size()) {
+		if ((++HC_current_file) >= (int)HC_preset_filenames.size()) {
```

Comment:
This should use a static_cast<int>. Or you could also just SCP_vector_inbounds for a more complete check.

Reply:
> Or you could also just SCP_vector_inbounds for a more complete check
Possible, but when using this it's important to make the increment of the index before the check

Reply:
Sounds good, updated with static_cast for now